tests/system/datastore/test_v3io.py

```diff
@@ -86,36 +141,225 @@ def test_v3io_large_object_upload(self, tmp_path):
+        assert response.decode() == self.test_string[20:]
+        response = data_item.get(size=20)
+        assert response.decode() == self.test_string[:20]
+        response = data_item.get(offset=20, size=0)
+        assert response.decode() == self.test_string[20:]
+        response = data_item.get(offset=20, size=10)
+        assert response.decode() == self.test_string[20:30]
+
+        with tempfile.NamedTemporaryFile(mode="w+", delete=True) as temp_file:
+            data_item.download(temp_file.name)
+            content = temp_file.read()
+            assert content == self.test_string
+
+        # append=True test:
+        data_item.put(self.test_string, append=True)
+        response = data_item.get()
+        assert response.decode() == self.test_string + self.test_string
+
+    def test_stat(self):
+        data_item, _ = self._get_data_item()
+        data_item.put(self.test_string)
+        stat = data_item.stat()
+        assert stat.size == len(self.test_string)
 
     def test_list_dir(self):
-        dir_base_path = "/bigdata/test_base_dir/"
-        v3io_dir_url = "v3io://" + dir_base_path
-        dir_base_item = mlrun.datastore.store_manager.object(v3io_dir_url)
-        file_item = mlrun.datastore.store_manager.object(v3io_dir_url + "test_file")
+        dir_base_item = mlrun.datastore.store_manager.object(self.object_dir_url)
+        file_item = mlrun.datastore.store_manager.object(
+            self.object_dir_url + "/test_file.txt"
+        )
         file_item_deep = mlrun.datastore.store_manager.object(
-            v3io_dir_url + "test_dir/test_file"
+            self.object_dir_url + "/test_dir/test_file.txt"
         )
         try:
             file_item.put("test")
             file_item_deep.put("test")
             actual_dir_content = dir_base_item.listdir()
-            assert actual_dir_content == ["test_dir/", "test_file"]
+            assert actual_dir_content == ["test_dir/", "test_file.txt"]
         finally:
             dir_base_item.delete()
+
+    def test_upload(self):
+        data_item, _ = self._get_data_item()
+        data_item.upload(self.test_file_path)
+        response = data_item.get()
+        assert response.decode() == self.test_string
+
+    def test_rm(self):
+        data_item, _ = self._get_data_item()
+        data_item.upload(self.test_file_path)
+        data_item.stat()
+        data_item.delete()
+        with pytest.raises(
+            mlrun.errors.MLRunNotFoundError, match="Request failed with status 404"
+        ):
+            data_item.stat()
+
+    @pytest.mark.skip(
+        reason="need to add support for recursive deletion in all datastores"
+    )
+    def test_dir_rm(self):
+        url = f"{self.object_dir_url}/test_directory/file.txt"
+        data_item = mlrun.get_dataitem(url=url)
+        data_item.upload(self.test_file_path)
+        dir_data_item = mlrun.get_dataitem(url=os.path.dirname(url))
+        dir_data_item.delete(recursive=True)
```

Comment:
Why not actually add this parameter and forward it? Instead of skipping the test.

Reply:
fixed